Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py

```diff
@@ -34,75 +37,73 @@
+                 resource_group_name: str, workspace_name: str,
+                 verify: bool = True, proxy: bool = False):
+
+        if not server_url:
+            server_url = f'https://management.azure.com/subscriptions/{subscription_id}/'\
+                         f'resourceGroups/{resource_group_name}/providers/Microsoft.OperationalInsights/workspaces/'\
+                         f'{workspace_name}/providers/Microsoft.SecurityInsights'
+
+        super().__init__(
             tenant_id=tenant_id,
-            auth_code=auth_code,
-            ok_codes=(200, 201, 202, 204, 400, 401, 403, 404)
+            auth_id=client_id,
+            enc_key=client_secret,
+            self_deployed=True,
+            grant_type=CLIENT_CREDENTIALS,
+            token_retrieval_url=TOKEN_RETRIEVAL_URI,
+            base_url=server_url,
+            scope=Scopes.management_azure,
+            ok_codes=(200, 201, 202, 204),
+            verify=verify,
+            proxy=proxy
         )
 
-    def http_request(self, method, url_suffix=None, full_url=None, params=None, data=None, is_get_entity_cmd=False):
-        if not params:
-            params = {}
+    def http_request(self, method, url_suffix=None, full_url=None, params=None, data=None):
         if not full_url:
+            params = params or {}
             params['api-version'] = API_VERSION
 
-        res = self.ms_client.http_request(method=method,  # disable-secrets-detection
-                                          url_suffix=url_suffix,
-                                          full_url=full_url,
-                                          json_data=data,
-                                          params=params,
-                                          resp_type='response')
-        res_json = res.json()
-
-        if res.status_code in (400, 401, 403, 404):
-            code = res_json.get('error', {}).get('code', 'Error')
-            error_msg = res_json.get('error', {}).get('message', res_json)
-            if res.status_code == 404 and is_get_entity_cmd:
-                error_msg += ENTITIES_RETENTION_PERIOD_MESSAGE
-            raise ValueError(
-                f'[{code} {res.status_code}] {error_msg}'
-            )
+        res = super().http_request(
+            method=method,  # disable-secrets-detection
+            url_suffix=url_suffix,
+            full_url=full_url,
+            headers={'User-Agent': XSOAR_USER_AGENT},
+            json_data=data,
+            params=params,
+            error_handler=error_handler,
+            resp_type='response')
 
-        return res_json
+        if res.content:
+            return res.json()
 
 
 ''' INTEGRATION HELPER METHODS '''
 
 
+def error_handler(response: requests.Response):
+    """
+    raise informative exception in case of error response
+    """
+    if response.status_code in (400, 401, 403, 404):
+        res_json = response.json()
+        code = res_json.get('error', {}).get('code', 'Error')
```

Comment:
What are the options for this variable?
I think that we need a better variable name to make it more descriptive.